src/Picasso_ParticleInit.hpp

```diff
@@ -55,19 +59,20 @@ void filterEmpties( const ExecutionSpace& exec_space,
             {
                 if ( final_pass )
                 {
-                    empties( count ) = i;
+                    empties( count ) = i + previous_num_particles;
                 }
                 ++count;
```

Comment:
Furthermore, is it true that the final value of `count` should be equal to `std::min(num_particles - previous_num_particles - local_num_create, local_num_create)`?

Reply:
It should be `local_num_create` by the end